src/core/appinterface.cpp

```diff
@@ -227,6 +231,60 @@ QVariantMap AppInterface::availableLanguages() const
   return languages;
 }
 
+void AppInterface::changeLanguage( const QString &languageCode )
+{
+  QSettings settings;
+  settings.setValue( QStringLiteral( "/customLanguage" ), languageCode );
```

Comment:
Is the languageCode is not empty, I'd do a check against available languages first to make sure we switch to something we can translate.